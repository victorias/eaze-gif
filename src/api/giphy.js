// @flow

export const getTrending = (apiKey: string, offset?: number = 0) => {
  return `https://api.giphy.com/v1/gifs/trending?api_key=${apiKey}&limit=25&rating=G&offset=${offset}`;
};

export const search = (apiKey: string, query: string, offset?: number = 0) => {
  return `https://api.giphy.com/v1/gifs/search?api_key=${apiKey}&q=${query}&limit=25&offset=0&rating=G&lang=en`;
};
