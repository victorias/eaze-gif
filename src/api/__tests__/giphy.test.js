// @flow

import * as giphyApi from '../giphy';

describe('giphy api tests', () => {
  it('.getTrending', () => {
    expect(giphyApi.getTrending('apiKey')).toMatchSnapshot();
    expect(giphyApi.getTrending('apiKey', 10)).toMatchSnapshot();
  });
});
