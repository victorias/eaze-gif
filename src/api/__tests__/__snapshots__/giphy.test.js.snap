// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`giphy api tests .getTrending 1`] = `"https://api.giphy.com/v1/gifs/trending?api_key=apiKey&limit=25&rating=G&offset=0"`;

exports[`giphy api tests .getTrending 2`] = `"https://api.giphy.com/v1/gifs/trending?api_key=apiKey&limit=25&rating=G&offset=10"`;
